src/pages/swap/index.tsx

```diff
@@ -109,14 +124,44 @@ export const SwapPage = () => {
       return;
     }
 
-    console.log('starting ....');
-
-    handleOnSubmit({
-      inputTokenType: from as InputTokenType,
-      outputTokenType: to as OutputTokenType,
-      amountInUnits: fromAmountString,
-      minAmountOutUnits: minimumOutputAmount.preciseString,
-    });
+    // test the route for starting token, then proceed
+    // will disable the confirm button
+    setIsInitiating(true);
+
+    const outputToken = OUTPUT_TOKEN_CONFIG[to];
+    const inputToken = INPUT_TOKEN_CONFIG[from];
+
+    // both route and stellar vault checks must be valid to proceed
+    const outputAmountBigMargin = Big(minimumOutputAmount.preciseString)
+      .round(2, 0)
+      .mul(1 + SPACEWALK_REDEEM_SAFETY_MARGIN); // add an X percent margin to be sure
+    const expectedRedeemAmountRaw = multiplyByPowerOfTen(outputAmountBigMargin, outputToken.decimals).toFixed();
+
+    const inputAmountBig = Big(fromAmountString);
+    const inputAmountRaw = multiplyByPowerOfTen(inputAmountBig, inputToken.decimals).toFixed();
+
+    Promise.all([
+      getVaultsForCurrency(
+        api!,
+        outputToken.stellarAsset.code.hex,
+        outputToken.stellarAsset.issuer.hex,
+        expectedRedeemAmountRaw,
+      ),
+      testRoute(fromToken, inputAmountRaw),
```

Comment:
We didn't specify it but should we also add a little margin to this test and on top of `inputAmountRaw`?

Reply:
I'll use the same `spacewalk` margin of 5% to avoid adding more constants if that's okay.